experiment/tracer/static/script.js

```diff
@@ -0,0 +1,273 @@
+'use strict';
+
+var logger = {
+  data: [],
+  message: '',
+  options: {
+    request: {
+      endpoint: 'https://' + window.location.hostname + '/trace?',
+      params: '',
+      config: {}
+    },
+    messages: {
+      invalidInput: 'Invalid input. Please pass a link from a PR or PR comment and profit',
+      invalidHost: 'Invalid host. Host needs to be github.com',
+      invalidParams: 'Invalid params in the URL. Need either "pr", "repo", and "org", or  "issuecomment" to be specified',
+      invalidUrl: 'Invalid link provided. Failed to construct \'URL\'',
+      requestError: 'Fetch error: Status is not 2xx',
+      requestLoading: 'Loading..',
+      emptyLogs: 'No logs to display'
+    }
+  },
+
+  // the actual fetch
+  request: function(endpoint) {
+    generateView.toggleLoader(true);
+    fetch(endpoint).then(
+        function(response) {
+
+          if (response.status !== 200) {
+            generateView.toggleLoader(false);
+            controller.setMessage(logger.options.messages.requestError);
+            return;
+          }
+
+          response.json().then(function(data) {
+            generateView.toggleLoader(false);
+            controller.setData(data);
+          });
+        }
+    ).catch(function(err) {
+      console.log('Fetch Error', err);
+      generateView.toggleLoader(false);
+      controller.setMessage(err);
+    });
+
+  }
+};
+
+var controller = {
+
+  init: function() {
+    generateView.init();
+  },
+
+  // Getter & Setters
+
+  setParams: function(params) {
+    logger.options.request.params = params;
+  },
+
+  setData: function(data) {
+    logger.data = data;
+    // render this view (update the DOM elements with the updated values)
+    generateView.render();
+  },
+
+  getParams: function() {
+    return logger.options.request.endpoint + encodeURI(logger.options.request.params);
+  },
+
+  setMessage: function(msg) {
+    logger.message = msg;
+    generateView.updateMsg();
+  },
+
+  getMessage: function() {
+    return logger.message;
+  },
+
+  // fetch data
+  requestLogs: function() {
+    //  params needed
+    return logger.request(this.getParams());
+  },
+
+  buildParams: function(userValue) {
```

Comment:
In general it would be nice to show the user what's wrong in their input in the error messages here. For instance, show which host they provided when telling them that GitHub is the only valid one, or show them the parameters when explaining that they have more than 5 tokens in the URL -- static messages are going to be more frustrating to understand.

Reply:
I changed it and add more dynamic error messages based on user's input. 
https://github.com/kubernetes/test-infra/pull/6241/commits/a6a09937d9c823ae567a4c777217129a7a537fe0#diff-7d200cf223d93ff3baf058b95c2106feR14